### ISSUE TITLE
[Bug] Unable to hide overlap in value labels in emphasis state

### ISSUE THREAD

**Issue by hughess** [apache/echarts#19192]
### Version

5.4.3

### Link to Minimal Reproduction

https://jsfiddle.net/s6mr0g8c/1/

### Steps to Reproduce

1. Create a multi-series line chart with many points
2. Set labelLayout to `hideOverlap`
3. Hover over one of the series names in the legend

### Current Behavior

On hovering the legend, labels for all points on the line appear, without any control for overlap

![CleanShot 2023-10-11 at 13 17 56](https://github.com/apache/echarts/assets/12602440/98f740e7-4a74-4126-91f1-adf0bafc8c31)

### Expected Behavior

Labels should stay the same as they are outside of emphasis state (or `hideOverlap` should apply in emphasis state)

### Environment

```markdown
N/A
```

**Comment by helgasoft** [comment]
the animated image shown does not correspond to the Minimal Reproduction code.
To fix the **code**, use ```emphasis: {disabled: true}``` on both series.
Please close issue if problem solved.

**Comment by hughess** [comment]
I don’t want to disable emphasis because id still like to be able to hover over the individual lines and have them be highlighted. I’d also like the line to be highlighted when you hover over the legend item - I just don’t want the labels to show up like that

**Comment by helgasoft** [comment]
Your title is correct, [labelLayout.hideOverlap](https://echarts.apache.org/en/option.html#series-line.labelLayout.hideOverlap) does not apply to emphasis state.  It could be a **new [Feature]** to implement.

Workaround to improve UI since both lines overlap:
second line: ```labelLayout: { hideOverlap: true, dy:22 },```
both lines:  ```emphasis: {focus:'self', label: {show: false}}```

**Comment by hughess** [comment]
Ok thanks. Let me know if I should change the issue title to request the feature - it would be helpful to have.

Unfortunately I'm building this into a library, so it needs to be useable for `n` lines that I won't know about in advance

**Comment by helgasoft** [comment]
yes, I think [Feature] title is more appropriate since no promises are broken.

**Comment by gaofanni** [comment]
`axisPointer.triggerEmphasis=false` can solve this problem

**Comment by helgasoft** [comment]
@gaofanni, ```axisPointer.triggerEmphasis``` exists already in the code but does not help.
The problem shows while hovering over the legend items.
